MsWheaPkg/MsWheaReport/Dxe/MsWheaReportList.c

```diff
@@ -54,6 +56,15 @@ CreateNewEntry (
   Index = 0;
   CopyMem (&((UINT8 *)MsWheaListEntry->PayloadPtr)[Index], MsWheaEntryMD, sizeof (MS_WHEA_ERROR_ENTRY_MD));
 
+  TempMsWheaEntryMD = (MS_WHEA_ERROR_ENTRY_MD *)MsWheaListEntry->PayloadPtr;
+  ExtraSectionPtr = (MS_WHEA_ERROR_EXTRA_SECTION_DATA *)(UINTN)(MsWheaEntryMD->ExtraSection);
+  if (ExtraSectionPtr != NULL) {
+    TempMsWheaEntryMD->ExtraSection = (EFI_PHYSICAL_ADDRESS)(UINTN)AllocateCopyPool (

```

Comment:
I didn't walk the logic all the way through, but does everything work out if `AllocateCopyPool()` fails here (returns `NULL`)?

Reply:
It would. This allocation contains its own data size and does not intrude into the normal packet reported, which is the normal case when other code reports telemetry without extra data.

Reply:
And that is checked here: https://github.com/microsoft/mu_plus/blob/4cbdbab1ef42919743c21d61106f775a55850d33/MsWheaPkg/MsWheaReport/MsWheaReportHER.c#L292

It will log other information but the extra data. I think that is probably the best we can do when it comes to out of resources error.